### DISCUSSION TITLE
Devnet Pre-Release 1.4.0alpha1

### DISCUSSION THREAD

**Discussion by lk86** [MinaProtocol/mina#12716] [Announcements]
The O(1) Labs team is proud to announce that 1.4.0alpha1 is ready to test on Devnet! This release includes a set of minor bugfixes, RFCs and new documentation, major testing and CI improvements, and code cleanup. The changes should have a minor effect on user experience but include months of work to improve testing, processes, and code quality.

If you run into any issues, please include `Release: 1.4.0alpha1` in the name of any issues you create. Please report any issues you encounter [here](https://github.com/MinaProtocol/mina/issues/new/choose) or join the Discussion below!

### Complete Changelog since 1.3.1.2:
https://github.com/MinaProtocol/mina/compare/1.3.1.2...1.4.0alpha1

### Debian Packages:
**Debian Repository:**
First, set up and update the `alpha` Debian Repository for your platform with the following commands:
```
sudo rm /etc/apt/sources.list.d/mina*.list
echo "deb [trusted=yes] http://packages.o1test.net $(lsb_release -cs) alpha" | sudo tee /etc/apt/sources.list.d/mina-alpha.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-devnet=1.4.0alpha1-da12bcf`
**Mina Archive:** `sudo apt-get install -y mina-archive=1.4.0alpha1-da12bcf`
**New Mina Archive Schema:** [create_schema.sql](https://github.com/MinaProtocol/mina/blob/da12bcf65fc6b37d81b237e57ab9284270d339ef/src/app/archive/create_schema.sql)

### Docker Images:
Every docker image is built for each of the supported platforms, replace `CODENAME` in the images below with your choice of `bionic`, `focal`, `stretch`, `buster`, or `bullseye` to select the base docker image.

**Mina Daemon:** `minaprotocol/mina-daemon:1.4.0alpha1-da12bcf-CODENAME-devnet`
**Mina Archive:** `minaprotocol/mina-archive:1.4.0alpha1-da12bcf-CODENAME`
**New Mina Archive Schema:** [create_schema.sql](https://github.com/MinaProtocol/mina/blob/da12bcf65fc6b37d81b237e57ab9284270d339ef/src/app/archive/create_schema.sql)
**Mina Rosetta:** `minaprotocol/mina-rosetta:1.4.0alpha1-da12bcf-CODENAME`

### Sandbox Node: For testing in an isolated, single-node network without snarks
As of 1.2.0, the demo functionality is built into the standard daemon container, via the environment variable `RUN_DEMO=true`.
Just `docker run -it --name mina-demo -e RUN_DEMO=true minaprotocol/mina-daemon:1.4.0alpha1-da12bcf-CODENAME-devnet` and go!

### Step by Step Guide:
Check out [our documentation](https://docs.minaprotocol.com/en/advanced/connecting-devnet) for complete instructions on using this version to connect to Devnet. 
*Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/seed-lists/devnet_seeds.txt`. or add `PEER_LIST_URL=https://storage.googleapis.com/seed-lists/devnet_seeds.txt` to your `.mina-env`.

If you are running the correct version on the correct network **DEVNET**, `mina client status` will show:
```
Chain id:     b6ee40d336f4cc3f33c1cc04dee7618eb8e556664c2b2d82ad4676b512a82418
Git SHA-1:    da12bcf65fc6b37d81b237e57ab9284270d339ef
```

<hr /><em>This discussion was created from the release <a href='https://github.com/MinaProtocol/mina/releases/tag/1.4.0alpha1'>Devnet Pre-Release 1.4.0alpha1</a>.</em>

**Comment by EmrePiconbello** [comment]
Couldn't find the image while we are trying to update our devnet node. We didn't see it here either https://hub.docker.com/r/minaprotocol/mina-daemon/tags but we might be looking to wrong place.

**Reply by psteckler** [reply]
> Couldn't find the image while we are trying to update our devnet node. We didn't see it here either [hub.docker.com/r/minaprotocol/mina-daemon/tags](https://hub.docker.com/r/minaprotocol/mina-daemon/tags) but we might be looking to wrong place.

`docker pull gcr.io/o1labs-192920/mina-daemon:da12bcf-bionic-devnet`
